activesupport/lib/active_support/cache/coder.rb

```diff
@@ -88,6 +81,36 @@ def load(dumped)
 
         MARSHAL_SIGNATURE = "\x04\x08".b.freeze
 
+        class LazyEntry < Cache::Entry
+          def initialize(serializer, compressor, payload, type, **options)
+            super(payload, **options)
+            @serializer = serializer
+            @compressor = compressor
+            @type = type
+          end
+
+          def value
+            if @type
+              @value = @compressor.inflate(@value) if @compressor
```

Comment:
I have a small worry here. If the serializer later raise for some reason, the lazy entry will be less with a decompressed `@value`.

I think we should store the decompressed payload in a local variable and only re-assing `@value` on both decompression and deserialization succeeded.